### PR TITLE
Add azure-quantum, qsharp, qdk

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BastianZim** [comment]
Waiting on https://github.com/microsoft/qdk-python/issues/207

File: recipes/azure-quantum/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% set name = "azure-quantum" %}
+{% set version = "0.21.2112.180703" %}
```
Comment (guenp) [inline recipes/azure-quantum/meta.yaml:2]:
```suggestion
{% set version = "0.22.187631" %}
```

File: recipes/qdk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{% set name = "qdk" %}
+{% set version = "0.21.2112.180703" %}
```
Comment (guenp) [inline recipes/qdk/meta.yaml:2]:
```suggestion
{% set version = "0.22.187631" %}
```

File: recipes/qsharp-chemistry/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{% set name = "qsharp-chemistry" %}
+{% set version = "0.21.2112.180703" %}
```
Comment (guenp) [inline recipes/qsharp-chemistry/meta.yaml:2]:
```suggestion
{% set version = "0.22.187631" %}
```

File: recipes/qsharp-core/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{% set name = "qsharp-core" %}
+{% set version = "0.21.2112.180703" %}
```
Comment (guenp) [inline recipes/qsharp-core/meta.yaml:2]:
```suggestion
{% set version = "0.22.187631" %}
```

File: recipes/qsharp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{% set name = "qsharp" %}
+{% set version = "0.21.2112.180703" %}
```
Comment (guenp) [inline recipes/qsharp/meta.yaml:2]:
```suggestion
{% set version = "0.22.187631" %}
```

**Comment by BastianZim** [comment]
@guenp Thanks for your help here!

I'm currently building the packages but it's failing because I'm only using the mock server since I only want to add the python recipes and not the whole package.

Error:
```shell
import: 'qsharp.chemistry'
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/qsharp-chemistry_1643650163957/test_tmp/run_test.py", line 2, in <module>
    import qsharp.chemistry
  File "/home/conda/staged-recipes/build_artifacts/qsharp-chemistry_1643650163957/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/qsharp/chemistry/__init__.py", line 40, in <module>
    qsharp.packages.add("microsoft.quantum.chemistry.jupyter")
  File "/home/conda/staged-recipes/build_artifacts/qsharp-chemistry_1643650163957/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/qsharp/packages.py", line 48, in add
    logger.info("Loading complete: " + ';'.join(str(e) for e in pkgs))
TypeError: 'NoneType' object is not iterable
```

Logs: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=450725&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1927

Do you know if there is a workaround, or do I need to actually build the whole package?

**Comment by BastianZim** [comment]
Closing this in favour of single PRs.

**Pull request closed without merging**